apm-commons/apm-datacarrier/src/main/java/org/apache/skywalking/apm/commons/datacarrier/common/AtomicRangeInteger.java

```diff
@@ -49,6 +49,18 @@ public final int getAndIncrement() {
         return current;
     }
 
+    public final int getAndIncrement() {
+        int next;
+        do {
+            next = this.value.incrementAndGet();
```

Comment:
From my understanding, you are using the `conflict abandon mechanism`, trade off more throughput. Because this line invokes, when comareAndSet fail, you can't get the index in this loop.

Please share with me, why you think this is better?  I think this is two sides of the coin
1. **[OLD]**, lower throughput, higher cache queue usage rate, less blank element.
1. **[NEW]**, opposite.

Reply:
The conflict in the New implementation only happens on the edge of the buffer, so the loop count in  New implementation is less than the Old one for the edge-off after abandoning conflict. It will produce  less blank element than the old one.
I can't understand what is "higher cache queue usage rate" in this case, could you please explain it ?

Reply:
> From my understanding, you are using the `conflict abandon mechanism`, trade off more throughput. Because this line invokes, when comareAndSet fail, you can't get the index in this loop.
> 
> Please share with me, why you think this is better? I think this is two sides of the coin
> 
> 1. **[OLD]**, lower throughput, higher cache queue usage rate, less blank element.
> 2. **[NEW]**, opposite.
The conflict in the New implementation only happens on the edge of the buffer, so the loop count in New implementation is less than the Old one for the edge-off after abandoning conflict. It will produce less blank element than the old one.
I can't understand what is "higher cache queue usage rate" in this case, could you please explain it ?

Reply:
Sorry, my fault, I misread the codes. Yes, the conflict only happens at the edge.